src/Pangraph.hs

```diff
@@ -1,116 +1,54 @@
 {-# LANGUAGE OverloadedStrings #-}
 
 module Pangraph (
+-- Abstract Types
 Pangraph,
-Node,
+MalformedEdge,
 Edge,
-Attribute,
-attributes,
-nodes,
-edges,
-key,
-value,
-pair,
-valuesByKey,
-makeAttribute,
-makeNode,
-makeEdge,
+Vertex,
+Attribute, -- A type alias for (Key, Value)
+Key,
+Value,
+VertexID,
+EdgeID,
+
+-- Constructors
 makePangraph,
-makeAlga
+makeEdge,
+makeVertex,
+
+-- Pangraph Getters
+edges,
+vertices,
+mapEdges,
+mapVertices,
+vertexByID,
+edgeByID,
+
+-- -- Operations on Pangraph
+-- addEdge,
+-- updateEdge,
+-- insertVertex,
+-- removeEdge,
+-- removeVertex,
+
+-- Getters on Vertex and Edge
+edgeAttributes,
+vertexAttributes,
+edgeEndpoints,
+edgeID,
+vertexID,
+lookupVertexValues,
+lookupEdgeValues,
+edgeContainsKey,
+vertexContainsKey,
+
+-- Utility Operations
+vertexToAssocList,
+edgeToAssocList
 ) where
 
+import Pangraph.Internal.Graph
```

Comment:
I remember you mentioned you had some issues with exporting the right functions. Have you added the internal module simply to solve the export problem? If yes, don't do that. Move everything to the top-level module.